From a2d2745c311baa588fb0fffbe38076294f06b7c0 Mon Sep 17 00:00:00 2001
From: Nicolai Buchwitz <n.buchwitz@kunbus.com>
Date: Wed, 12 Jul 2023 11:30:42 +0200
Subject: [PATCH] cfg80211: ship debian certificates as hex files

Loading the regulatory database from the debian files fails with

"loaded regulatory.db is malformed or signature is missing/invalid"

due to missing certificates.  Add these debian-specific certificates
from upstream to fix this error. See #5536 for details.

The certificates have been imported as following:

patch -p1 <<<$(
curl https://salsa.debian.org/kernel-team/linux/-/raw/\
master/debian/patches/debian/\
wireless-add-debian-wireless-regdb-certificates.patch
)

Signed-off-by: Nicolai Buchwitz <n.buchwitz@kunbus.com>
---
 net/wireless/certs/debian.hex | 1426 +++++++++++++++++++++++++++++++++
 1 file changed, 1426 insertions(+)
 create mode 100644 net/wireless/certs/debian.hex

--- /dev/null
+++ b/net/wireless/certs/debian.hex
@@ -0,0 +1,1426 @@
+0x30,
+0x82,
+0x02,
+0xbd,
+0x30,
+0x82,
+0x01,
+0xa5,
+0x02,
+0x14,
+0x57,
+0x7e,
+0x02,
+0x1c,
+0xb9,
+0x80,
+0xe0,
+0xe8,
+0x20,
+0x82,
+0x1b,
+0xa7,
+0xb5,
+0x4b,
+0x49,
+0x61,
+0xb8,
+0xb4,
+0xfa,
+0xdf,
+0x30,
+0x0d,
+0x06,
+0x09,
+0x2a,
+0x86,
+0x48,
+0x86,
+0xf7,
+0x0d,
+0x01,
+0x01,
+0x0b,
+0x05,
+0x00,
+0x30,
+0x1a,
+0x31,
+0x18,
+0x30,
+0x16,
+0x06,
+0x03,
+0x55,
+0x04,
+0x03,
+0x0c,
+0x0f,
+0x62,
+0x65,
+0x6e,
+0x68,
+0x40,
+0x64,
+0x65,
+0x62,
+0x69,
+0x61,
+0x6e,
+0x2e,
+0x6f,
+0x72,
+0x67,
+0x30,
+0x20,
+0x17,
+0x0d,
+0x32,
+0x30,
+0x30,
+0x31,
+0x33,
+0x30,
+0x31,
+0x33,
+0x32,
+0x36,
+0x31,
+0x33,
+0x5a,
+0x18,
+0x0f,
+0x32,
+0x31,
+0x32,
+0x30,
+0x30,
+0x31,
+0x30,
+0x36,
+0x31,
+0x33,
+0x32,
+0x36,
+0x31,
+0x33,
+0x5a,
+0x30,
+0x1a,
+0x31,
+0x18,
+0x30,
+0x16,
+0x06,
+0x03,
+0x55,
+0x04,
+0x03,
+0x0c,
+0x0f,
+0x62,
+0x65,
+0x6e,
+0x68,
+0x40,
+0x64,
+0x65,
+0x62,
+0x69,
+0x61,
+0x6e,
+0x2e,
+0x6f,
+0x72,
+0x67,
+0x30,
+0x82,
+0x01,
+0x22,
+0x30,
+0x0d,
+0x06,
+0x09,
+0x2a,
+0x86,
+0x48,
+0x86,
+0xf7,
+0x0d,
+0x01,
+0x01,
+0x01,
+0x05,
+0x00,
+0x03,
+0x82,
+0x01,
+0x0f,
+0x00,
+0x30,
+0x82,
+0x01,
+0x0a,
+0x02,
+0x82,
+0x01,
+0x01,
+0x00,
+0x9d,
+0xe1,
+0x77,
+0xa0,
+0x24,
+0xa0,
+0xd5,
+0x79,
+0x65,
+0x3a,
+0x07,
+0x90,
+0xc9,
+0xf6,
+0xa5,
+0xa6,
+0x1f,
+0x84,
+0x1c,
+0x23,
+0x07,
+0x4b,
+0x4f,
+0xa5,
+0x03,
+0xc6,
+0x0f,
+0xf7,
+0x54,
+0xd5,
+0x8b,
+0x7e,
+0x79,
+0x81,
+0x00,
+0xd2,
+0xe9,
+0x3d,
+0xf4,
+0x97,
+0xfe,
+0x84,
+0xcd,
+0x55,
+0xbd,
+0xc9,
+0x8f,
+0x21,
+0x57,
+0x88,
+0x06,
+0x39,
+0x90,
+0x66,
+0x41,
+0x26,
+0x79,
+0x2c,
+0xca,
+0x3f,
+0x95,
+0x87,
+0x01,
+0x11,
+0x2f,
+0x2f,
+0xb0,
+0xe1,
+0x0b,
+0x43,
+0xfc,
+0x5f,
+0x2f,
+0x4f,
+0x67,
+0x04,
+0xdb,
+0x4d,
+0xb7,
+0x72,
+0x4d,
+0xd1,
+0xc5,
+0x76,
+0x73,
+0x4d,
+0x91,
+0x69,
+0xb0,
+0x71,
+0x17,
+0x36,
+0xea,
+0xab,
+0x0a,
+0x3a,
+0xcd,
+0x95,
+0x9b,
+0x76,
+0x1b,
+0x8e,
+0x21,
+0x17,
+0x8f,
+0xc5,
+0x02,
+0xbf,
+0x24,
+0xc7,
+0xc0,
+0x40,
+0xb1,
+0x3b,
+0xc4,
+0x80,
+0x7c,
+0x71,
+0xa5,
+0x51,
+0xdc,
+0xf7,
+0x3a,
+0x58,
+0x7f,
+0xb1,
+0x07,
+0x81,
+0x8a,
+0x10,
+0xd1,
+0xf6,
+0x93,
+0x17,
+0x71,
+0xe0,
+0xfa,
+0x51,
+0x79,
+0x15,
+0xd4,
+0xd7,
+0x8f,
+0xad,
+0xbd,
+0x6f,
+0x38,
+0xe1,
+0x26,
+0x7d,
+0xbc,
+0xf0,
+0x3e,
+0x80,
+0x89,
+0xb4,
+0xec,
+0x8e,
+0x69,
+0x90,
+0xdb,
+0x97,
+0x8a,
+0xf0,
+0x23,
+0x23,
+0x83,
+0x82,
+0x3b,
+0x6a,
+0xb1,
+0xac,
+0xeb,
+0xe7,
+0x99,
+0x74,
+0x2a,
+0x35,
+0x8e,
+0xa9,
+0x64,
+0xfd,
+0x46,
+0x9e,
+0xe8,
+0xe5,
+0x48,
+0x61,
+0x31,
+0x6e,
+0xe6,
+0xfc,
+0x19,
+0x18,
+0x54,
+0xc3,
+0x1b,
+0x4f,
+0xd6,
+0x00,
+0x44,
+0x87,
+0x1c,
+0x37,
+0x45,
+0xea,
+0xf5,
+0xc9,
+0xcb,
+0x0f,
+0x0c,
+0x55,
+0xec,
+0xcf,
+0x6a,
+0xc2,
+0x45,
+0x26,
+0x23,
+0xa2,
+0x31,
+0x52,
+0x4d,
+0xee,
+0x21,
+0x7d,
+0xfd,
+0x58,
+0x72,
+0xc2,
+0x28,
+0xc5,
+0x8e,
+0xa9,
+0xd0,
+0xee,
+0x01,
+0x77,
+0x08,
+0xa5,
+0xf0,
+0x22,
+0x2b,
+0x47,
+0x79,
+0x2b,
+0xcf,
+0x9a,
+0x46,
+0xb5,
+0x8f,
+0xfd,
+0x64,
+0xa2,
+0xb5,
+0xed,
+0x02,
+0x03,
+0x01,
+0x00,
+0x01,
+0x30,
+0x0d,
+0x06,
+0x09,
+0x2a,
+0x86,
+0x48,
+0x86,
+0xf7,
+0x0d,
+0x01,
+0x01,
+0x0b,
+0x05,
+0x00,
+0x03,
+0x82,
+0x01,
+0x01,
+0x00,
+0x20,
+0x44,
+0xfe,
+0xa9,
+0x9e,
+0xdd,
+0x9b,
+0xea,
+0xce,
+0x25,
+0x75,
+0x08,
+0xf0,
+0x2b,
+0x53,
+0xf7,
+0x5a,
+0x36,
+0x1c,
+0x4a,
+0x23,
+0x7f,
+0xd0,
+0x41,
+0x3c,
+0x12,
+0x2b,
+0xb9,
+0x80,
+0x4e,
+0x8a,
+0x15,
+0x5d,
+0x1f,
+0x40,
+0xa7,
+0x26,
+0x28,
+0x32,
+0xc3,
+0x5b,
+0x06,
+0x28,
+0x2d,
+0x3d,
+0x08,
+0x09,
+0x1e,
+0x01,
+0xe9,
+0x67,
+0xe3,
+0x33,
+0xe6,
+0x15,
+0x45,
+0x39,
+0xee,
+0x17,
+0x83,
+0xdb,
+0x42,
+0xff,
+0x7f,
+0x35,
+0xf4,
+0xac,
+0x16,
+0xdb,
+0xba,
+0xb8,
+0x1a,
+0x20,
+0x21,
+0x41,
+0xff,
+0xf3,
+0x92,
+0xff,
+0x65,
+0x6e,
+0x29,
+0x16,
+0xd0,
+0xbf,
+0x8d,
+0xdf,
+0x48,
+0x2c,
+0x73,
+0x36,
+0x7f,
+0x22,
+0xe6,
+0xee,
+0x78,
+0xb4,
+0x63,
+0x83,
+0x0e,
+0x39,
+0xeb,
+0xaf,
+0x10,
+0x2a,
+0x90,
+0xd3,
+0xfc,
+0xe6,
+0xc3,
+0x8f,
+0x97,
+0x5b,
+0x76,
+0xbf,
+0x9b,
+0xf5,
+0x98,
+0xd2,
+0x53,
+0x06,
+0x8b,
+0xf8,
+0xa4,
+0x04,
+0x9b,
+0x1b,
+0x62,
+0x6a,
+0x9d,
+0xac,
+0xe6,
+0x4b,
+0x0d,
+0xc9,
+0xd7,
+0x56,
+0x63,
+0x15,
+0x01,
+0x38,
+0x8c,
+0xbe,
+0xf1,
+0x44,
+0xc4,
+0x38,
+0x27,
+0xe0,
+0xcf,
+0x72,
+0xd6,
+0x3d,
+0xe4,
+0xf7,
+0x4b,
+0x3b,
+0xd2,
+0xb1,
+0x0c,
+0xd5,
+0x83,
+0x6d,
+0x1e,
+0x10,
+0x04,
+0x69,
+0x29,
+0x88,
+0x69,
+0xe0,
+0x7d,
+0xd7,
+0xdb,
+0xb4,
+0x59,
+0x72,
+0x8d,
+0x9d,
+0x3c,
+0x43,
+0xaf,
+0xc6,
+0x7d,
+0xb7,
+0x21,
+0x15,
+0x52,
+0x8a,
+0xe9,
+0x9b,
+0x6b,
+0x2e,
+0xe8,
+0x27,
+0x3c,
+0x3f,
+0x2d,
+0x84,
+0xfb,
+0x9a,
+0x22,
+0x0a,
+0x9f,
+0x6a,
+0x25,
+0xe6,
+0x39,
+0xe4,
+0x74,
+0x73,
+0xb6,
+0x2a,
+0x70,
+0xaa,
+0x1d,
+0xcb,
+0xcc,
+0xd4,
+0xa0,
+0x1b,
+0x26,
+0x71,
+0x63,
+0x04,
+0xc5,
+0x12,
+0x21,
+0x48,
+0xba,
+0x92,
+0x27,
+0x06,
+0xa8,
+0x3e,
+0x6d,
+0xa1,
+0x43,
+0xa5,
+0xd2,
+0x2a,
+0xf7,
+0xca,
+0xc4,
+0x26,
+0xe8,
+0x5b,
+0x1f,
+0xe4,
+0xdc,
+0x89,
+0xdc,
+0x1f,
+0x04,
+0x79,
+0x3f,
+0x30,
+0x82,
+0x02,
+0xcd,
+0x30,
+0x82,
+0x01,
+0xb5,
+0x02,
+0x14,
+0x3a,
+0xbb,
+0xc6,
+0xec,
+0x14,
+0x6e,
+0x09,
+0xd1,
+0xb6,
+0x01,
+0x6a,
+0xb9,
+0xd6,
+0xcf,
+0x71,
+0xdd,
+0x23,
+0x3f,
+0x03,
+0x28,
+0x30,
+0x0d,
+0x06,
+0x09,
+0x2a,
+0x86,
+0x48,
+0x86,
+0xf7,
+0x0d,
+0x01,
+0x01,
+0x0b,
+0x05,
+0x00,
+0x30,
+0x22,
+0x31,
+0x20,
+0x30,
+0x1e,
+0x06,
+0x03,
+0x55,
+0x04,
+0x03,
+0x0c,
+0x17,
+0x72,
+0x6f,
+0x6d,
+0x61,
+0x69,
+0x6e,
+0x2e,
+0x70,
+0x65,
+0x72,
+0x69,
+0x65,
+0x72,
+0x40,
+0x67,
+0x6d,
+0x61,
+0x69,
+0x6c,
+0x2e,
+0x63,
+0x6f,
+0x6d,
+0x30,
+0x20,
+0x17,
+0x0d,
+0x32,
+0x30,
+0x30,
+0x32,
+0x32,
+0x34,
+0x31,
+0x39,
+0x30,
+0x31,
+0x34,
+0x34,
+0x5a,
+0x18,
+0x0f,
+0x32,
+0x31,
+0x32,
+0x30,
+0x30,
+0x31,
+0x33,
+0x31,
+0x31,
+0x39,
+0x30,
+0x31,
+0x34,
+0x34,
+0x5a,
+0x30,
+0x22,
+0x31,
+0x20,
+0x30,
+0x1e,
+0x06,
+0x03,
+0x55,
+0x04,
+0x03,
+0x0c,
+0x17,
+0x72,
+0x6f,
+0x6d,
+0x61,
+0x69,
+0x6e,
+0x2e,
+0x70,
+0x65,
+0x72,
+0x69,
+0x65,
+0x72,
+0x40,
+0x67,
+0x6d,
+0x61,
+0x69,
+0x6c,
+0x2e,
+0x63,
+0x6f,
+0x6d,
+0x30,
+0x82,
+0x01,
+0x22,
+0x30,
+0x0d,
+0x06,
+0x09,
+0x2a,
+0x86,
+0x48,
+0x86,
+0xf7,
+0x0d,
+0x01,
+0x01,
+0x01,
+0x05,
+0x00,
+0x03,
+0x82,
+0x01,
+0x0f,
+0x00,
+0x30,
+0x82,
+0x01,
+0x0a,
+0x02,
+0x82,
+0x01,
+0x01,
+0x00,
+0xf0,
+0xb8,
+0x4f,
+0x3f,
+0x70,
+0x78,
+0xf8,
+0x74,
+0x45,
+0xa2,
+0x28,
+0xaf,
+0x04,
+0x75,
+0x04,
+0xa3,
+0xf3,
+0xa7,
+0xc7,
+0x04,
+0xac,
+0xb6,
+0xe1,
+0xfc,
+0xe1,
+0xc0,
+0x3d,
+0xe0,
+0x26,
+0x90,
+0x8a,
+0x45,
+0x60,
+0xc4,
+0x75,
+0xf3,
+0x1a,
+0x33,
+0x37,
+0x56,
+0x7d,
+0x30,
+0x07,
+0x75,
+0x0e,
+0xa6,
+0x79,
+0x06,
+0x95,
+0x9d,
+0x17,
+0x3c,
+0x09,
+0xa9,
+0x7f,
+0xab,
+0x95,
+0x5d,
+0xed,
+0xe0,
+0x75,
+0x26,
+0x2f,
+0x65,
+0x65,
+0xcd,
+0x61,
+0xb1,
+0x33,
+0x27,
+0x67,
+0x41,
+0xa1,
+0x01,
+0x13,
+0xe9,
+0x13,
+0x6a,
+0x6d,
+0x4e,
+0x98,
+0xe1,
+0x9e,
+0x7b,
+0x0b,
+0x5b,
+0x44,
+0xef,
+0x68,
+0x5a,
+0x6f,
+0x7d,
+0x97,
+0xa1,
+0x33,
+0x22,
+0x97,
+0x12,
+0x21,
+0x09,
+0x8f,
+0x90,
+0xe0,
+0x25,
+0x94,
+0xdd,
+0x8a,
+0x3a,
+0xf7,
+0x4a,
+0x60,
+0x04,
+0x26,
+0x6d,
+0x00,
+0x82,
+0xe4,
+0xcf,
+0x64,
+0x1c,
+0x79,
+0x15,
+0x24,
+0xf2,
+0x42,
+0x86,
+0xf5,
+0x10,
+0x86,
+0xac,
+0x20,
+0x88,
+0x90,
+0x87,
+0xdf,
+0x8c,
+0x37,
+0x7c,
+0xbf,
+0x35,
+0xd5,
+0x6f,
+0x9f,
+0x77,
+0xc3,
+0xcd,
+0x69,
+0x25,
+0x06,
+0xc2,
+0x65,
+0x51,
+0x71,
+0x89,
+0x7f,
+0x6e,
+0x4d,
+0xe5,
+0xd5,
+0x8a,
+0x36,
+0x1a,
+0xad,
+0xc1,
+0x18,
+0xd6,
+0x14,
+0x42,
+0x87,
+0xf0,
+0x93,
+0x83,
+0xf1,
+0x99,
+0x74,
+0xc4,
+0x13,
+0xaa,
+0x3b,
+0x66,
+0x85,
+0x6f,
+0xe0,
+0xbc,
+0x5f,
+0xb6,
+0x40,
+0xa6,
+0x41,
+0x06,
+0x0a,
+0xba,
+0x0e,
+0xe9,
+0x32,
+0x44,
+0x10,
+0x39,
+0x53,
+0xcd,
+0xbf,
+0xf3,
+0xd3,
+0x26,
+0xf6,
+0xb6,
+0x2b,
+0x40,
+0x2e,
+0xb9,
+0x88,
+0xc1,
+0xf4,
+0xe3,
+0xa0,
+0x28,
+0x77,
+0x4f,
+0xba,
+0xa8,
+0xca,
+0x9c,
+0x05,
+0xba,
+0x88,
+0x96,
+0x99,
+0x54,
+0x89,
+0xa2,
+0x8d,
+0xf3,
+0x73,
+0xa1,
+0x8c,
+0x4a,
+0xa8,
+0x71,
+0xee,
+0x2e,
+0xd2,
+0x83,
+0x14,
+0x48,
+0xbd,
+0x98,
+0xc6,
+0xce,
+0xdc,
+0xa8,
+0xa3,
+0x97,
+0x2e,
+0x40,
+0x16,
+0x2f,
+0x02,
+0x03,
+0x01,
+0x00,
+0x01,
+0x30,
+0x0d,
+0x06,
+0x09,
+0x2a,
+0x86,
+0x48,
+0x86,
+0xf7,
+0x0d,
+0x01,
+0x01,
+0x0b,
+0x05,
+0x00,
+0x03,
+0x82,
+0x01,
+0x01,
+0x00,
+0x76,
+0x5d,
+0x03,
+0x3d,
+0xb6,
+0x96,
+0x00,
+0x1b,
+0x6e,
+0x0c,
+0xdd,
+0xbb,
+0xc8,
+0xdf,
+0xbc,
+0xeb,
+0x6c,
+0x01,
+0x40,
+0x1a,
+0x2b,
+0x07,
+0x60,
+0xa1,
+0x1a,
+0xe1,
+0x43,
+0x57,
+0xfa,
+0xbe,
+0xde,
+0xbb,
+0x8f,
+0x73,
+0xf3,
+0x92,
+0xa2,
+0xaa,
+0x83,
+0x01,
+0xc1,
+0x17,
+0xe4,
+0x9d,
+0x09,
+0x41,
+0xe0,
+0x32,
+0x33,
+0x97,
+0x4b,
+0xf2,
+0xdc,
+0x0f,
+0x8b,
+0xa8,
+0xb8,
+0x5a,
+0x04,
+0x86,
+0xf6,
+0x71,
+0xa1,
+0x97,
+0xd0,
+0x54,
+0x56,
+0x10,
+0x8e,
+0x54,
+0x99,
+0x0d,
+0x2a,
+0xa9,
+0xaf,
+0x1b,
+0x55,
+0x59,
+0x06,
+0x2b,
+0xa4,
+0x5f,
+0xb1,
+0x54,
+0xa6,
+0xec,
+0xc7,
+0xd6,
+0x43,
+0xee,
+0x86,
+0x2c,
+0x9b,
+0x18,
+0x9d,
+0x8f,
+0x00,
+0x82,
+0xc1,
+0x88,
+0x61,
+0x16,
+0x85,
+0x3c,
+0x17,
+0x56,
+0xfe,
+0x6a,
+0xa0,
+0x7a,
+0x68,
+0xc5,
+0x7b,
+0x3d,
+0x3c,
+0xb6,
+0x13,
+0x18,
+0x99,
+0x6d,
+0x74,
+0x65,
+0x13,
+0x67,
+0xb7,
+0xfc,
+0x5a,
+0x44,
+0x48,
+0x72,
+0xa0,
+0x73,
+0xb8,
+0xff,
+0x02,
+0x9d,
+0x7c,
+0x5b,
+0xf9,
+0x7c,
+0x75,
+0x0a,
+0x3c,
+0x81,
+0x80,
+0x3c,
+0x41,
+0xf2,
+0xd5,
+0xfa,
+0x3d,
+0x1f,
+0xe3,
+0xda,
+0x8c,
+0xa5,
+0x17,
+0x1f,
+0x53,
+0x1a,
+0x75,
+0xad,
+0x4e,
+0x11,
+0x1c,
+0x07,
+0xec,
+0x0a,
+0x69,
+0xfd,
+0x33,
+0xfa,
+0x32,
+0x7e,
+0x66,
+0xf5,
+0x29,
+0xe8,
+0x4d,
+0x8a,
+0xfa,
+0x0d,
+0x4b,
+0x68,
+0xc3,
+0x95,
+0x11,
+0xba,
+0x6f,
+0x1e,
+0x07,
+0x8c,
+0x85,
+0xc7,
+0xc7,
+0xc9,
+0xc1,
+0x30,
+0xa3,
+0x70,
+0xb0,
+0xa1,
+0xe0,
+0xd5,
+0x85,
+0x15,
+0x94,
+0x77,
+0xc1,
+0x1c,
+0x91,
+0xf1,
+0x5f,
+0x50,
+0xcd,
+0x2c,
+0x57,
+0x4b,
+0x22,
+0x4f,
+0xee,
+0x95,
+0xd7,
+0xa7,
+0xa4,
+0x59,
+0x62,
+0xae,
+0xb9,
+0xbf,
+0xd7,
+0x63,
+0x5a,
+0x04,
+0xfc,
+0x24,
+0x11,
+0xae,
+0x34,
+0x4b,
+0xf4,
+0x0c,
+0x9f,
+0x0b,
+0x59,
+0x7d,
+0x27,
+0x39,
+0x54,
+0x69,
+0x4f,
+0xfd,
+0x6e,
+0x44,
+0x9f,
+0x21,
